api-service/src/services/DatasetService.ts

```diff
@@ -26,10 +27,12 @@ export const getDraftDataset = async (dataset_id: string) => {
     return DatasetDraft.findOne({ where: { dataset_id }, raw: true });
 }
 
-export const setApiId = async (req: any, apiId: string) => {
-    return _.set(req, "id", apiId)
-}
-
 export const getDraftTransformations = async (dataset_id: string) => {
     return DatasetTransformationsDraft.findAll({ where: { dataset_id }, raw: true });
 }
+
+export const setReqDatasetId = (req: Request, dataset_id: string) => {
+    if (dataset_id) {
+        return _.set(req, "dataset_id", dataset_id)
```

Comment:
if present set otherwise set to null

Reply:
Yes am setting it while getting the metric labels

const getMetricLabels = (req: any, res: Response) => {
     const { id, entity, url, startTime } = req;
     const { statusCode = 200 } = res
     const request_size = req.socket.bytesRead
     const response_size = res.getHeader('content-length');
     **const dataset_id = _.get(req, "dataset_id") || null**
     const duration = getDuration(startTime);
     const metricLabels = { entity, id, endpoint: url, dataset_id, status: statusCode, request_size, response_size }
     return { duration, metricLabels }
 }